File: Android/PalindromeApp/palindrome/src/main/java/com/aferdita/palindrome/Palindrome.kt
package com.aferdita.palindrome

/**
 * Created by aferditamuriqi on 11/17/17.
 */

fun isPalindrome(testString:String): Pair<Boolean, String> {

    val myAlphaNumericString = testString.replace("[^A-Za-z0-9]".toRegex(), "").toLowerCase()

    var left = 0
    var right = myAlphaNumericString.length - 1

    while (left < right)
    {
        if (myAlphaNumericString[left] != myAlphaNumericString[right])
        {
            return Pair(false , myAlphaNumericString)
        }

        left += 1
        right -= 1
    }
    return Pair(true , myAlphaNumericString)
}

File: iOS/Palindrome/Palindrome/Palindrome.swift
//
//  Palindrome.swift
//  Palindrome
//
//  Created by Aferdita Muriqi on 11/17/17.
//  Copyright © 2017 DITA. All rights reserved.
//

import Foundation

public func isPalindrome(_ testString:String) -> (status:Bool,result:String) {
  let charSet = CharacterSet.alphanumerics.inverted
  let myAlphaNumericString = testString.components(separatedBy: charSet).joined(separator: "").uppercased() as NSString
  
  var left = 0
  var right = myAlphaNumericString.length - 1
  
  while left < right
  {
    if myAlphaNumericString.character(at: left) != myAlphaNumericString.character(at: right)
    {
      return (false, myAlphaNumericString as String)
    }
    
    left += 1
    right -= 1
  }
  
  return (true,myAlphaNumericString as String)
}
